### PR TITLE
feat(graphql): archive page integration

### PR DESCRIPTION
**Pull request by 120EE0692**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [ ] Make sure you are making a pull request against the **default branch** (left side). Also you should start _your branch_ off _default branch_.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.
- [ ] I have added necessary documentation (if appropriate)

### Linked Issues

Fixes

### Description

Please describe your pull request.

❤️Thank you!

### Post merge checklist

- [ ] Follow steps from the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: client/src/config/ApolloClient.js [hunk 1]
```diff
@@ -40,8 +40,8 @@ const link = from([
   new HttpLink({
     uri:
       process.env.NODE_ENV === 'production'
-        ? 'https://mm.dashnet.in/api/v1/graph'
-        : 'https://project-reclamation-staging.herokuapp.com/v1/graph',
+        ? 'http://localhost:5000/v1/graph'
+        : 'http://localhost:5000/v1/graph',
```
Comment (rutajdash) [inline client/src/config/ApolloClient.js:44]: Why this change?

File: client/src/graphql/queries/article/listArticleByYearAndMonth.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { gql } from '@apollo/client';
+
+import basicArticleData from '../../fragments/article/basicArticleData';
+
+const listArticlesByYearAndMonth = gql`
+    query ListArticlesByYearAndMonth($onlyPublished: Boolean!, $limit: Int, $offset: Int, $year: Int, $month: Int) {
+  listArticlesByYearAndMonth(onlyPublished: $onlyPublished, limit: $limit, offset: $offset, year: $year, month: $month) {
+        ${basicArticleData}
+  }
+}
```
Comment (rutajdash) [inline client/src/graphql/queries/article/listArticleByYearAndMonth.js:10]: Please rectify the indentation in this GQL code.

File: client/src/pages/archive/[...yearAndMonth].jsx [hunk 3]
```diff
@@ -72,10 +81,61 @@ function ArchivePage({}) {
...
+    variables: {
+      onlyPublished: true,
+      year: parseInt(year),
+      month: ARCHIVES.months.indexOf(month),
+      limit: 9,
+    },
+  });
+  return {
+    props: { archiveArticles, year, month },
+    revalidate: 7 * 24 * 60 * 60,
```
Comment (rutajdash) [inline client/src/pages/archive/[...yearAndMonth].jsx:118]: Put a condition here for the revalidation time to be 1 hour or so during Monday and Tuesday, and 5 days for the rest. It would make more sense as we make many changes during the first few days of the publishing.

File: client/src/pages/archive/[...yearAndMonth].jsx [hunk 4]
```diff
@@ -72,10 +81,61 @@ function ArchivePage({}) {
...
+    .map((year) =>
+      months.map((month) => {
+        return {
+          params: {
+            yearAndMonth: [`${year}`, month],
+          },
+        };
+      }),
+    )
+    .flat();
```
Comment (rutajdash) [inline client/src/pages/archive/[...yearAndMonth].jsx:135]: Why are we prebuilding all these pages? Is it necessary? Why not just build the latest year and allow the rest to be built in ISR.

**Review by rutajdash — CHANGES_REQUESTED** [review]

File: client/src/pages/archive/[...yearAndMonth].jsx (hunk 4 above)
Comment (120EE0692) [reply client/src/pages/archive/[...yearAndMonth].jsx:135]: No it's fine.

File: client/src/pages/archive/[...yearAndMonth].jsx (hunk 3 above)
Comment (120EE0692) [reply client/src/pages/archive/[...yearAndMonth].jsx:118]: Na this is also fine, archive me articles ek week bad jayega to thik rahega.

**Review by rutajdash — APPROVED** [review]
